### PR TITLE
Use multithreaded assigner

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

**Review by akokoshn — CHANGES_REQUESTED** [review]
Please check parallel-crypto3 headers carefully.
Any idea how to check automatically i used right headers? May be  some static-assert?

**Comment by AndreyMlashkin** [comment]
> Please check parallel-crypto3 headers carefully. Any idea how to check automatically i used right headers? May be some static-assert?

I don't think it's possible. Because, come of the includes should still point to the single-threaded implementation.

File: parallel-crypto3/libs/parallel-containers/include/nil/parallel-crypto3/container/accumulation_vector.hpp
[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallel-containers/include/nil/parallel-crypto3/container/accumulation_vector.hpp:1]: done

File: proof-producer/bin/proof-producer/CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +47,7 @@ function(setup_proof_generator_target)
     if(ARG_ADDITIONAL_DEPENDENCIES)
         foreach(lib IN LISTS ARG_ADDITIONAL_DEPENDENCIES)
             target_link_libraries(${ARG_TARGET_NAME}-lib INTERFACE ${lib})
+            target_compile_definitions(${ARG_TARGET_NAME}-lib INTERFACE MULTITHREADED)
```
Comment (akokoshn) [inline proof-producer/bin/proof-producer/CMakeLists.txt:50]: check if `lib` is `parallel-crypto3`.
Generally single-thread version also can has some additional dependency

**Review by akokoshn — CHANGES_REQUESTED** [review]
What about includes in transpiler and zkevm-frameowrk ((should use #ifdef MULTITHREAD) )

**Pull request closed without merging**